tests/utils/delivery-start-date.spec.js

```diff
@@ -0,0 +1,104 @@
+const DeliveryStartDate = require('../../utils/delivery-start-date');
+const expect = require('chai').expect;
+const fetchMock = require('fetch-mock');
+const sinon = require('sinon');
+
+describe('DeliveryStartDate', () => {
+	let sandbox = sinon.createSandbox();
+
+	let document;
+	let startDateContainer;
+	let startDateFieldStub;
+	let startDateTextStub;
+
+	beforeEach(() => {
+		document = { querySelector: sandbox.stub().returns(false) };
+
+		startDateContainer = { classList: { add: sandbox.stub(), remove: sandbox.stub() } };
+		startDateFieldStub = { value: '2019-02-16' };
+		startDateTextStub = { innerHTML: 'Saturday 16th of February 2019' };
+
+		document.querySelector.withArgs('#deliveryStartDateField').returns(startDateContainer);
+		document.querySelector.withArgs('#deliveryStartDate').returns(startDateFieldStub);
+		document.querySelector.withArgs('.js-start-date-text').returns(startDateTextStub);
+	});
+
+	afterEach(() => {
+		fetchMock.restore();
+		sandbox.restore();
+	});
+
+	describe('constructor', () => {
+		it('should throw an error if document element isn\'t passed in.', () => {
+			expect(() => {
+				new DeliveryStartDate();
+			}).to.throw();
+		});
+
+		it('should throw an error if delivery start date element does not exist on the page', () => {
+			expect(() => {
+				document.querySelector = () => { };
+				new DeliveryStartDate(document);
+			}).to.throw();
+		});
+	});
+
+	describe('handleDeliveryStartDateChange', () => {
+		let startDateUtil;
+		let startDateChangeResult;
+
+		async function setup () {
+			fetchMock.mock('/foo/bar', {
+				firstDeliveryDate: '2019-04-13',
+				firstDeliveryDateString: 'Saturday 13th of April 2019'
+			});
+			startDateUtil = new DeliveryStartDate(document);
+			startDateChangeResult = await startDateUtil.handleDeliveryStartDateChange('/foo/bar', () => {
```

Comment:
```suggestion
			startDateChangeResult = await startDateUtil.handleDeliveryStartDateChange('/api/path', () => {
```
Shall we have something more descriptive, I find using foo/bar everywhere makes it hard to read. 
In this case, we both have foo/bar in the url and the returned payload just below.